app/controllers/smart_proxies_controller.rb

```diff
@@ -45,6 +45,12 @@ def tftp_server
     requested_data(:tftp_server)
   end
 
+  def puppet_environments
+    render :partial => 'smart_proxies/plugins/puppet_envs', :locals => {:envs => @smart_proxy.environment_stats}
+  rescue Foreman::Exception => exception
+    process_ajax_error exception
```

Comment:
The messages rendered when the proxy isn't reachable don't look great - 
![screenshot from 2016-01-18 20-33-48](https://cloud.githubusercontent.com/assets/598891/12401813/41104ff4-be29-11e5-9233-cfb0d2967d74.png)

Same goes for PuppetCA & Autosign